resource/plugins/unreal/python/loader/finalizers/unreal_asset_loader_finalizer.py

```diff
@@ -0,0 +1,171 @@
+    '''Unreal asset load finalizer plugin.'''
+
+    plugin_name = 'unreal_asset_loader_finalizer'
+
+    def run(self, context_data=None, data=None, options=None):
+        '''Restore asset info.'''
+
+        result = {}
+
+        # First, evaluate asset path
+        asset_filesystem_path = None
+        for comp in data:
+            if comp['type'] == core_constants.COMPONENT:
+                for result in comp['result']:
+                    if result['type'] == core_constants.IMPORTER:
+                        plugin_result = result['result'][0]
+                        asset_filesystem_path = list(
+                            plugin_result['result']['run_result'].values()
+                        )[0]
+                        break
+
+        if asset_filesystem_path is None:
+            # No asset path found, abort
+            return {'message': 'No loaded asset path found.'}
+
+        # Expect: "C:\\Users\\Henrik Norin\\Documents\\Unreal Projects\\MyEmptyProject\\Content\\Assets\\NewWorld.umap"
+        # Transform to asset path
+        asset_path = unreal_utils.filesystem_asset_path_to_asset_path(
+            asset_filesystem_path
+        )
+
+        # Load the asset in Unreal
+        self.logger.debug(
+            'Result of loading asset "{}" in Unreal editor: {}'.format(
+                asset_path, unreal.EditorAssetLibrary.load_asset(asset_path)
+            )
+        )
+
+        # Restore pipeline asset info
+        assetversion = self.session.query(
+            'AssetVersion where id is "{}"'.format(context_data['version_id'])
+        ).one()
+        ident = str_version(assetversion, by_task=False)
+        metadata = None
+        if 'ftrack-connect-pipeline-unreal' in assetversion['metadata']:
+            metadata = json.loads(
+                assetversion['metadata']['ftrack-connect-pipeline-unreal']
+            )
+            if 'pipeline_asset_info' in metadata:
+                asset_info = metadata['pipeline_asset_info']
+
+                dependency_assetversion = self.session.query(
+                    'AssetVersion where id is "{}"'.format(
+                        asset_info[asset_const.VERSION_ID]
+                    )
+                ).one()
+                dependency_ident = str_version(
+                    dependency_assetversion, by_task=False
+                )
+                self.logger.debug(
+                    'Restoring pipeline asset info for {}: {}'.format(
+                        dependency_ident, asset_info
+                    )
+                )
+                ftrack_object_manager = UnrealFtrackObjectManager(
+                    self.event_manager
+                )
+                ftrack_object_manager.asset_info = asset_info
+                dcc_object = UnrealDccObject()
+                dcc_object.name = (
+                    ftrack_object_manager._generate_dcc_object_name()
+                )
+                if dcc_object.exists():
```

Comment:
Comment out what happen when a dcc_object already exists to point to the fbx file of the uasset and we are creating a new dcc_object for the same uasset. Botth will have the same name.

Reply:
Added a comment